week1.md

```diff
@@ -9,17 +9,172 @@ these things yet):
+| conversationId | ObjectId                 | N         | References Conversation.\_id |
+| role           | "owner" \| "participant" | N         |                              |
+
+## Task
+
+Create a database schema here which has all the entities which we need to implement
+the chat system. Hint: read the requirements of the system in the README and use
+that as a starting point for listing entities.
+
+### Messenger application schema
+
+## User schema
+
+| Attribute      | Type                     | Optional? | Other Notes                  |
+| -------------- | ------------------------ | --------- | ---------------------------- |
+| \_id           | ObjectId                 | N         | Primary key, Unique          |
+| email          | String                   | N         | Unique, email address       |
+| password 			 | String                   | N         | Encrypted, must contain special characters, numbers, and at least a capital letter  |
+| contactList    | Object 									| N         | Here will be saved all of the user's contacts|
+| role           | "owner" \| "list"				| N 				| 															| 
+| fullName 			 | String 									| Y 				| Optional full name of the user. Can be left empty for anonymity |
+
+## ContactList schema
+
+| Attribute      | Type                     | Optional? | Other Notes                  |
+| -------------- | ------------------------ | --------- | ---------------------------- |
+| \_id           | ObjectId                 | N         | Primary key, Unique          |
+| owner          | Object                   | N         | User who owns this contact list |
+| contacts 			 | List                   	| N         | List of contacts the user has  |
+| contactList    | Object 									| N         | Here will be saved all of the user's contacts|
+| role           | "list"	\| "owner"  			| N 				| 															| 
+
+## Contact scheme
+
+| Attribute      | Type                     | Optional? | Other Notes                  |
+| -------------- | ------------------------ | --------- | ---------------------------- |
+| \_id           | ObjectId                 | N         | Primary key, Unique          |
+| list           | Object                   | N         | List where this contact is saved in |
+| isConversationStarted | boolean           | N         | Whether the user has started a conversation with this contact  |
+| sentMessages   | List 										| N         | A list of messages sent by the user to the contact |
+| receivedMessages | List                   | N 	 	 		| A list of messages sent by the contact to the user | 
+| role           | "contact" \| "list"      | N         |                              | 
+
+## Message scheme
+
+| Attribute      | Type                     | Optional? | Other Notes                  |
+| -------------- | ------------------------ | --------- | ---------------------------- |
+| \_id           | ObjectId                 | N         | Primary key, Unique          |
```

Comment:
I'd also recommend adding an `authorId` attribute to this schema to track who sent the message:

```suggestion
| \_id           | ObjectId                 | N         | Primary key, Unique          |
| authorId           | ObjectId                 | N         | References User.\_id          |
```